### PR TITLE
Add DateTimeUtc scalar

### PR DESCRIPTION
**Pull request by spawnia**
- [x] Added or updated tests
- [x] Added Docs for all relevant versions
- [x] Updated CHANGELOG.md

#1255 

**Changes**

@Christophvh

**Breaking changes**

### CODE REVIEW

**Comment by spawnia** [comment]
@Christophvh i had trouble with parsing that format, i am not sure if i got it right.

**Comment by Christophvh** [comment]
> @Christophvh i had trouble with parsing that format, i am not sure if i got it right.

Yes, but I think the parse method should not enforce a format. Any valid date is fine imo, because there are many variations for the ISO-format and when we receive those they should also work. Also a front-end can send a date without a time-zone, in that case, we can assume this follows the timezone of the application.

Carbon::parse(); will work.

**Comment by spawnia** [comment]
`Carbon::parse()` will accept just about anything you throw at it, which i don't think is too great of an idea. Garbage in -> Garbage out.

I do think we should enforce UTC, but might have to allow some degree of freedom in regards to sub-second measurements. Browsers have restricted the precision you can get in order to prevent spectre, meltdown.

```
(new Date).toJSON()
"2020-04-24T08:21:17.625Z"
```

**Comment by Christophvh** [comment]
> `Carbon::parse()` will accept just about anything you throw at it (e.g. `Monday next week`), which i don't think is too great of an idea. Garbage in -> Garbage out.
> 
> I do think we should enforce UTC, but might have to allow some degree of freedom in regards to sub-second measurements. Browsers have restricted the precision you can get in order to prevent spectre, meltdown.
> 
> ```
> (new Date).toJSON()
> "2020-04-24T08:21:17.625Z"
> ```

Yes true, maybe something like this:  Carbon::createFromTimeString('17:00:00', 'UTC');

File: src/Schema/Types/Scalars/DateTimeTz.php [hunk 1]
```diff
@@ -8,11 +8,11 @@ class DateTimeTz extends DateScalar
...
     protected function format(Carbon $carbon): string
     {
-        return $carbon->toIso8601String();
+        return $carbon->toJSON();
     }
 
     protected function parse($value): Carbon
     {
-        return Carbon::createFromFormat(Carbon::ISO8601, $value);
+        return Carbon::createFromIsoFormat('YYYY-MM-DDTHH:mm:ss.SSSSSSZ', $value);
```
Comment (Christophvh) [inline src/Schema/Types/Scalars/DateTimeTz.php:16]:
```suggestion
        return Carbon::createFromTimeString($value, 'UTC');
```

**Pull request merged**